### ISSUE TITLE
Considerations for import absolute-first when using ES2020 Dynamic imports? 

### ISSUE THREAD

**Issue by joshchernoff** [benmosher/eslint-plugin-import#1955]
I would assume the use case for dynamic imports. 

IE:
```
import("foo").then( foo => {} )
```
Would be more likely used throughout the body of the document as opposed to being constrained to the head of the document considering the nature of code splitting. 

Would it be possible to ignore import first in cases where import is followed by then?  

Going off of:
https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/first.md

**Comment by ljharb** [comment]
I’m not sure what you mean - only static imports should be reported by this rule, and I’m not sure how code splitting (an invisible implementation detail of bundlers) would be relevant.

**Comment by joshchernoff** [comment]
This is how this came to my attention. 
<img width="523" alt="app_js_—_morphic_pro" src="https://user-images.githubusercontent.com/961877/100147223-e9847c00-2e4f-11eb-9929-f7307ba889f5.png">

In this use case I'm given an eslint error of 
`Parsing error: 'import' and 'export' may only appear at the top level`
For the given import in the screen cap.

**Comment by ljharb** [comment]
That’s not an eslint rule error - that just means your config isn’t set up to support parsing dynamic imports.

**Comment by joshchernoff** [comment]
"config isn’t set up to support parsing dynamic imports." 
I assume you mean the eslint config? 
Where can I learn to configure that then?

**Comment by ljharb** [comment]
Yes. You’ll need an ecmaVersion of 2020 in parserOptions, or to use babel-eslint as your parser.

**Comment by joshchernoff** [comment]
Thanks for the help, that looks to have addressed my concerns.
